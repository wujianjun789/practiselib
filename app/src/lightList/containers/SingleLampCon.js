/**
 * Created by Azrael on 2017/9/1.
 */
import '../../../public/styles/lightManage-list.less';
import React, { Component } from 'react';
import Content from '../../components/Content';
import SearchText from '../../components/SearchText';
import Select from '../../components/Select.1';
import TableWithHeader from '../components/TableWithHeader';
import TableTr from '../components/TableTr';
import Page from '../../components/Page';
import { getDomainList } from '../../api/domain';
import { getSearchAssets, getSearchCount, getDeviceStatusByModelAndId, updateAssetsRpcById } from '../../api/asset';
import { getLightLevelConfig } from '../../util/network';
import { getMomentDate, momentDateFormat } from '../../util/time';
import { message } from 'antd'
import 'antd/lib/message/style/css';
export default class SingleLampCon extends Component {
    constructor(props) {
        super(props);
        this.state = {
            page: {
                total: 0,
                current: 1,
                limit: 10
            },
            search: {
                value: '',
                placeholder: '请输入设备名称',
            },
            sidebarCollapse: false,
            currentDevice: null,
            deviceList: [],
            currentDomain: null,
            domainList: {
                titleField: 'name',
                valueField: 'name',
                options: []
            },
            currentSwitchStatus: 'on',
            deviceSwitchList: {
                titleField: 'title',
                valueField: 'value',
                options: [
                    { value: 'on', title: '开启' },
                    { value: 'off', title: '关闭' }
                ]
            },
            currentBrightness: 0,
            brightnessList: {
                titleField: 'title',
                valueField: 'value',
                options: []
            },
        };

        this.model = 'lc';

        this.columns = [
            { accessor: 'name', title: '设备名称' },
            { accessor: 'online', title: '在线状态' },
            { accessor: 'device', title: '灯状态  ' },
            { accessor: 'switch', title: '开关状态' },
            { accessor: 'brightness', title: '亮度    ' },
            { accessor: 'voltage', title: '电压    ' },
            { accessor: 'current', title: '电流    ' },
            { accessor: 'power', title: '功率    ' },
            { accessor: 'totalPower', title: '总电能  ' },
            { accessor: 'runningTime', title: '运行时间' },
            { accessor: 'lightTime', title: '亮灯时间' },
            {
                accessor(data) {
                    return data.updateTime ? momentDateFormat(getMomentDate(data.updateTime, 'YYYY-MM-DDTHH:mm:ss Z'), 'YYYY/MM/DD HH:mm') : '';
                },
                title: '更新时间'
            }
        ];

        this.collapseHandler = this.collapseHandler.bind(this);
        this.onChange = this.onChange.bind(this);
        this.pageChange = this.pageChange.bind(this);
        this.searchChange = this.searchChange.bind(this);
        this.searchSubmit = this.searchSubmit.bind(this);
        this.tableClick = this.tableClick.bind(this);

        this.initData = this.initData.bind(this);
        this.updateDomainData = this.updateDomainData.bind(this);
        this.initDeviceData = this.initDeviceData.bind(this);
        this.updateDeviceData = this.updateDeviceData.bind(this);
        this.updatePageSize = this.updatePageSize.bind(this);
        this.updateBrightnessList = this.updateBrightnessList.bind(this);
    }
    componentWillMount() {
        this.mounted = true;
        this.initData();
    }

    componentWillUnmount() {
        this.mounted = false;
    }

    initData() {
        getDomainList((data) => {
            this.mounted && this.updateDomainData(data, this.initDeviceData);
        });
        getLightLevelConfig(this.updateBrightnessList);
    }

    updateDomainData(data, cb) {
        let currentDomain,
            options = data;
        if (data.length == 0) {
            currentDomain = null;
        } else {
            currentDomain = data[0];
        };
        this.setState({ domainList: { ...this.state.domainList, options }, currentDomain }, () => {
            cb && cb()
        });
    }

    initDeviceData(isSearch) {

        const { search: { value }, page, currentDomain } = this.state;
        if (isSearch) {
            page.current = 1;
            this.setState({ page: page });
        }
        const { limit, current } = this.state.page;
        const offset = limit * (current - 1);
        getSearchAssets(currentDomain ? currentDomain.id : null, this.model, value, offset, limit, this.mounted && this.updateDeviceData);
        getSearchCount(currentDomain ? currentDomain.id : null, this.model, value, this.mounted && this.updatePageSize);

    }

    updateDeviceData(data) {
        let currentDevice = data.length == 0 ? null : data[0];
        this.setState({ deviceList: data, currentDevice });
    }

    updatePageSize(data) {
        this.setState({ page: { ...this.state.page, total: data.count } })
    }

    updateBrightnessList(data) {
        // ["关", 0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100]
        let opt = [];
        data.shift();   // 删除"关"
        data.forEach(value => {
            let val = value;
            let title = `亮度${val}`;
            opt.push({ value: val, title });
        });
        this.setState({ brightnessList: Object.assign({}, this.state.brightnessList, { options: opt }) });
    }

    onChange(e) {
        const { id, value } = e.target;
        switch (id) {
            case 'domain':
                let currentDomain = this.state.domainList.options[e.target.selectedIndex];
                this.setState({ currentDomain }, this.initDeviceData);
                break;
            case 'deviceSwitch':
                this.setState({ currentSwitchStatus: value });
                break;
            case 'dimming':
                this.setState({ currentBrightness: value });
                break;
        }
    }

    pageChange(page) {
        this.setState({ page: { ...this.state.page, current: page } }, this.initDeviceData);
    }

    searchChange(value) {
        this.setState({
            search: { ...this.state.search, value: value }
        })
    }

    searchSubmit() {
        this.initDeviceData(true);
    }

    collapseHandler() {
        this.setState({ sidebarCollapse: !this.state.sidebarCollapse });
    }

    tableClick(currentDevice) {
        this.setState({ currentDevice });
    }
    switchApply = () => {
        const { id } = this.state.currentDevice;
        const { currentSwitchStatus } = this.state;
        updateAssetsRpcById(id, { "status": currentSwitchStatus }, res => {
            if (res.success) {
                message.success('操作成功')
            } else {
                message.error('操作失败')
            }
        })
    }
    dimmingApply = () => {
        const { id } = this.state.currentDevice;
        const { currentBrightness } = this.state;
        updateAssetsRpcById(id, { "brightness": currentBrightness }, res => {
            if (res.success) {
                message.success('操作成功')
            } else {
                message.error('操作失败')
            }
        })
    }
    render() {
        const { page: { total, current, limit }, sidebarCollapse, currentDevice, deviceList,
            search: { value, placeholder }, currentDomain, domainList, deviceSwitchList,
            brightnessList, currentSwitchStatus, currentBrightness } = this.state;

        const disabled = deviceList.length == 0 ? true : false;
        return <Content className={`list-lc ${sidebarCollapse ? 'collapse' : ''}`}>
            <div className="content-left">
                <div className="heading">
                    <Select id='domain' titleField={domainList.titleField} valueField={domainList.valueField} options={domainList.options}
                        value={currentDomain == null ? '' : currentDomain[this.state.domainList.valueField]} onChange={this.onChange} />
                    <SearchText placeholder={placeholder} value={value} onChange={this.searchChange} submit={this.searchSubmit} />
                </div>
                <div className="table-container">
                    <TableWithHeader columns={this.columns}>
                        {
                            deviceList.map(item => <TableTr key={item.id} data={item} columns={this.columns} activeId={currentDevice.id}
                                rowClick={this.tableClick} willMountFuncs={[getDeviceStatusByModelAndId(this.model, item.id)]}
                            />)
                        }
                    </TableWithHeader>
                    <Page className={`page ${total == 0 ? "hidden" : ''}`} showSizeChanger pageSize={limit}
                        current={current} total={total} onChange={this.pageChange} />
                </div>
            </div>
            <div className={`container-fluid sidebar-info ${sidebarCollapse ? "sidebar-collapse" : ""}`}>
                <div className="row collapse-container" onClick={this.collapseHandler}>
                    <span className={sidebarCollapse ? "icon_horizontal" : "icon_vertical"}></span>
                </div>
                <div className="panel panel-default panel-1">
                    <div className="panel-heading">
                        <span className="icon_select"></span>选中设备
                            </div>
                    <div className="panel-body">
                        <span title={currentDevice == null ? '' : currentDevice.name}>{currentDevice == null ? '' : currentDevice.name}</span>
                    </div>
                </div>
                <div className="panel panel-default panel-2">
                    <div className="panel-heading">
                        <span className="icon_touch"></span>设备操作
                            </div>
                    <div className="panel-body">
                        <div>
                            <span className="tit">设备开关：</span>
                            <Select id="deviceSwitch" titleField={deviceSwitchList.titleField} valueField={deviceSwitchList.valueField}
                                options={deviceSwitchList.options} value={currentSwitchStatus} onChange={this.onChange} disabled={disabled} />
                            <button className="btn btn-primary" disabled={disabled} onClick={this.switchApply}>应用</button>
                        </div>
                        <div>
                            <span className="tit">调光：</span>
                            <Select id="dimming" titleField={brightnessList.titleField} valueField={brightnessList.valueField}
                                options={brightnessList.options} value={currentBrightness} onChange={this.onChange} disabled={disabled} />
                            <button className="btn btn-primary" disabled={disabled} onClick={this.dimmingApply}>应用</button>
                        </div>
                    </div>
                </div>
            </div>
        </Content>
    }
}

/**
 *                            <Table columns={this.columns} keyField='id' data={deviceList} rowClick={this.tableClick}
                                activeId={currentDevice == null ? '' : currentDevice.id}/>
 */
