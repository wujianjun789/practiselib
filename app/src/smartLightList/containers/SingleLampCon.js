/**
 * Created by Azrael on 2017/9/1.
 */
import '../../../public/styles/smartLightManage-list.less';
import React, {Component} from 'react';
import Content from '../../components/Content';
import SearchText from '../../components/SearchText';
import Select from '../../components/Select.1';
import Table from '../../components/Table2';
import Page from '../../components/Page';
export default class SingleLampCon extends Component {
    constructor(props) {
        super(props);
        this.state = {
            page: {
                total: 0,
                current: 1,
                limit: 10
            },
            search: {
                value: '',
                placeholder: '请输入设备名称',

            },
            sidebarCollapse: false,
            data: [],
            selectDevice: {

            }
        };

        this.columns = [
            {field: 'name', title: '策略名称'},
            {field: 'sensorType', title: '传感器类型'}
        ];

        this.collpseHandler = this.collpseHandler.bind(this);
        this.pageChange = this.pageChange.bind(this);
        this.searchChange = this.searchChange.bind(this);

        this.initData = this.initData.bind(this);
    }

    componentWillMount() {
        this.mounted = true;
        this.initData();
    }

    componentWillUnmount() {
        this.mounted = false;
    }
    
    initData() {

    }

    pageChange(page) {
        this.setState({page: {...this.state.page, current: page}}, this.initData);
    }

    searchChange(value) {
        this.setState({
            search: {...this.state.search, value: value}
        })
    }

    collpseHandler(){
        this.setState({sidebarCollapse: !this.state.sidebarCollapse});
    }
  
    render() {
        const {page: {total, current, limit}, sidebarCollapse, data, search: {value, placeholder}} = this.state;
        return <Content className={`list-lc ${sidebarCollapse ? 'collapse' : ''}`}>
                    <div className="content-left">
                        <div className="heading">
                            <Select />
                            <Select />
                            <SearchText placeholder={placeholder} value={value} onChange={this.searchChange}/>
                        </div>
                        <div className="body">
                            <Table columns={this.columns} keyField='id' data={data} isEdit rowEdit={this.tableRowEdit} rowDelete={this.tableRowDelete} />
                            <Page className={`page ${total==0?"hidden":''}`} showSizeChanger pageSize={limit}
                                current={current} total={total} onChange={this.pageChange}/>
                        </div>
                    </div>
                    <div className={`container-fluid sidebar-info ${sidebarCollapse ? "sidebar-collapse" : ""}`}>
                        <div className="row collapse-container" onClick={this.collpseHandler}>
                            <span className={sidebarCollapse ? "icon_horizontal"  :"icon_verital"}></span>
                        </div>
                        <div className="panel panel-default">
                            <div className="panel-heading">
                                <span className="icon_sys_select"></span>选中设备
                            </div>
                            <div className="panel-body">
                                <span className="domain-name">{'To be or not to be'}</span>
                            </div>
                        </div>
                    </div>
                </Content>
    }
}