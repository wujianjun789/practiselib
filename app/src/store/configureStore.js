import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../common/reducer';
import thunk from 'redux-thunk';


export default function configureStore(initialState) {

    const composeEnhancers =
        process.env.NODE_ENV !== 'production' &&
            typeof window === 'object' &&
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
                // Specify here name, actionsBlacklist, actionsCreators and other options
            }) : compose;

    const enhancer = composeEnhancers(
        applyMiddleware(thunk),
        // other store enhancers if any
    );
    const store = createStore(rootReducer, initialState, enhancer);
    return store;

}
