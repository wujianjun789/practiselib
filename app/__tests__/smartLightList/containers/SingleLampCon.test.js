/**
 * Created by Azrael on 2017/9/27
 */
jest.mock('../../../src/util/network.js');
jest.mock('../../../src/api/asset.js');
jest.mock('../../../src/api/domain.js');
import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl-redux';
import configureStore from '../../../src/store/configureStore';

import SingleLampCon from '../../../src/smartLightList/containers/SingleLampCon';

const store = configureStore();
describe('<SingleLampCon />', () => {
	it('default render', () => {
		const root = mount(<Provider store={store}>
				<IntlProvider>
					<SingleLampCon />
				</IntlProvider>
			</Provider>);
		const cmp = root.find('SingleLampCon');
		const content = cmp.find('Content');
		expect(!content.hasClass('collapse')).toBeTruthy();

		// const select = cmp.find('#domain');
		// const domainList_state = cmp.state('domainList');
		// const currentDomain_state = cmp.state('currentDomain');
		// Object.keys(domainList_state).forEach(item => {
		// 	expect(select.prop(item)).toEqual(domainList_state[item]);
		// });
		// expect(select.prop('value')).toEqual(currentDomain_state == null ? '' : currentDomain_state[domainList_state.valueField]);
        //
		// const searchText = cmp.find('SearchText');
		// const search_state = cmp.state('search');
		// Object.keys(search_state).forEach(item => {
		// 	expect(searchText.prop(item)).toBe(search_state[item]);
		// });

		// const tableWithHeader = cmp.find('TableWithHeader');
		// const inst = root.instance();
		// const columns = inst.columns;
		// expect(tableWithHeader.prop('columns')).toEqual(columns);

		// const tableTrs = cmp.find('TableTr');
		// const deviceList_state = root.state('deviceList');
		// const currentDevice_state = root.state('currentDevice');
		// deviceList_state.forEach((item, index) => {
		// 	let curTr = tableTrs.at(index);
		// 	expect(curTr.key()).toBe(item.id);
		// 	expect(curTr.prop('data')).toEqual(item);
		// 	expect(curTr.prop('columns')).toEqual(columns);
		// 	expect(curTr.prop('activeId')).toBe(currentDevice_state.id);
		// });

		// const page = cmp.find('Page');
		// const page_state = root.state('page');
		// expect(page.hasClass('hidden')).toBe(page_state.total == 0 ? true : false);
		// expect(page.prop('pageSize')).toBe(page_state.limit);
		// expect(page.prop('current')).toBe(page_state.current);
		// expect(page.prop('total')).toBe(page_state.total);

		const sidebarInfo = cmp.find('.sidebar-info');
		expect(!sidebarInfo.hasClass('sidebar-collapse')).toBeTruthy();

		// 测试icon className
		expect(sidebarInfo.find('.icon_vertical').length).toBe(1);

		const panel1 = sidebarInfo.find('.panel-1');
		// expect(panel1.find('.panel-heading').text()).toBe('选中设备');
		// expect(panel1.find('.panel-body span').text()).toBe(currentDevice_state == null ? '' : currentDevice_state.name);
		// expect(panel1.find('.panel-body span').props().title).toBe(currentDevice_state == null ? '' : currentDevice_state.name);

		const panel2 = sidebarInfo.find('.panel-2');
		// expect(panel2.find('.panel-heading').text()).toBe('设备操作');
		const panelBody = panel2.find('.panel-body');
		expect(panelBody.children('div').length).toBe(2);

		// expect(panelBody.childAt(0).find('.tit').text()).toBe('设备开关：');
		// const deviceSwitchList_state = root.state('deviceSwitchList');
		// Object.keys(deviceSwitchList_state).forEach(item => {
		// 	expect(panelBody.childAt(0).find('Select').prop(item)).toEqual(deviceSwitchList_state[item]);
		// });
		// const currentSwitchStatus_state = root.state('currentSwitchStatus');
		// expect(panelBody.childAt(0).find('Select').prop('value')).toBe(currentSwitchStatus_state);
		// expect(panelBody.childAt(0).find('Select').prop('disabled')).toBe(deviceList_state.length == 0 ? true : false);
		// // expect(panelBody.childAt(0).find('button').text()).toBe('应用');
		// expect(panelBody.childAt(0).find('button').prop('disabled')).toBe(deviceList_state.length == 0 ? true : false);

		// expect(panelBody.childAt(1).find('.tit').text()).toBe('调光：');
		// const brightnessList_state = root.state('brightnessList');
		// Object.keys(brightnessList_state).forEach(item => {
		// 	expect(panelBody.childAt(1).find('Select').prop(item)).toEqual(brightnessList_state[item]);
		// });
		// const currentBrightness_state = root.state('currentBrightness');
		// expect(panelBody.childAt(1).find('Select').prop('value')).toBe(currentBrightness_state);
		// expect(panelBody.childAt(1).find('Select').prop('disabled')).toBe(deviceList_state.length == 0 ? true : false);
		// // expect(panelBody.childAt(1).find('button').text()).toBe('应用');
		// expect(panelBody.childAt(1).find('button').prop('disabled')).toBe(deviceList_state.length == 0 ? true : false);
	});

	it('simulate click', () => {
		const root = mount(<Provider store={store}>
			<IntlProvider>
				<SingleLampCon />
			</IntlProvider>
		</Provider>);

		const cmp = root.find('SingleLampCon');
		let content = cmp.find('Content');
		expect(!content.hasClass('collapse')).toBeTruthy();
		content.find('.sidebar-info .collapse-container').simulate('click');
		content = cmp.find('Content');
		expect(content.hasClass('collapse')).toBeTruthy();
		expect(content.find('.sidebar-info').hasClass('sidebar-collapse')).toBeTruthy();
		expect(content.find('.sidebar-info .collapse-container span').hasClass('icon_horizontal')).toBeTruthy();
		content.find('.sidebar-info .collapse-container').simulate('click');
		content = cmp.find('Content');

		const domainList = {
			titleField: 'name',
			valueField: 'name',
			options: [{id: 1, name: '域一'},{id: 2, name: '域二'}]
		}
		root.setState({domainList});
		const _select = cmp.find('#domain');
		let event = {target: {id: 'domain', selectedIndex: 0}};
		_select.find('select').simulate('change', event);
		expect(_select.prop('value')).toBe(domainList.options[event.target.selectedIndex].name);

		const searchText = cmp.find('SearchText');
		event = {target: {value: '搜索'}};
		searchText.find('input').simulate('change', event);
		expect(searchText.prop('value')).toBe(event.target.value);

		const deviceList = [
			{id: 0, name: '设备1', online: true},
			{id: 1, name: '设备2', online: false}
		];
		// const currentDevice = deviceList[0];
		// root.setState({deviceList, currentDevice});
		// const tableTrs = cmp.find('TableTr');
		// tableTrs.at(1).find('tr').simulate('click');
		// expect(root.state('currentDevice')).toEqual(deviceList[1]);

		let deviceSwitch = cmp.find('#deviceSwitch');
		event = {target: { id: 'deviceSwitch', value: 'off'}};
		deviceSwitch.find('select').simulate('change', event);
		deviceSwitch = cmp.find('#deviceSwitch');
		expect(deviceSwitch.prop('value')).toBe(event.target.value);

		let dimming = cmp.find('#dimming');
		event = {target: { id: 'dimming', value: '30'}};
		dimming.find('select').simulate('change', event);
		dimming = cmp.find('#dimming');
		expect(dimming.prop('value')).toBe(event.target.value);


	});

	it('default render snapshot', () => {
		const cmp = renderer.create(<Provider store={store}>
			<IntlProvider>
				<SingleLampCon />
			</IntlProvider>
		</Provider>);
		expect(cmp.toJSON()).toMatchSnapshot();
	})
})
