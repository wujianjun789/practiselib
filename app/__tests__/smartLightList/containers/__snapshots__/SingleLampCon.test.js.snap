exports[`<SingleLampCon /> default render snapshot 1`] = `
<div
  className="content list-lc ">
  <div
    className="content-left">
    <div
      className="heading">
      <select
        className="form-control select "
        id="domain"
        onChange={[Function]}
        value="" />
      <div
        className="searchText "
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="请输入设备名称"
          type="search"
          value="" />
        <ul
          className="" />
        <span
          className="glyphicon glyphicon-search"
          onClick={[Function]} />
      </div>
    </div>
    <div
      className="table-container">
      <div
        className="table-responsive undefined">
        <table
          className="table table-hover">
          <thead>
            <tr>
              <th>
                设备名称
              </th>
              <th>
                在线状态
              </th>
              <th>
                灯状态
              </th>
              <th>
                开关状态
              </th>
              <th>
                亮度
              </th>
              <th>
                电压
              </th>
              <th>
                电流
              </th>
              <th>
                功率
              </th>
              <th>
                更新时间
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <ul
        className="rc-pagination page hidden"
        style={Object {}}
        unselectable="unselectable">
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="上一页">
          <a />
        </li>
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="下一页">
          <a />
        </li>
        <div
          className="rc-pagination-options" />
      </ul>
    </div>
  </div>
  <div
    className="container-fluid sidebar-info ">
    <div
      className="row collapse-container"
      onClick={[Function]}>
      <span
        className="icon_verital" />
    </div>
    <div
      className="panel panel-default panel-1">
      <div
        className="panel-heading">
        <svg>
          <use
            height="200"
            transform="scale(0.08,0.08)"
            viewBox="0 0 20 20"
            width="200"
            x="0"
            xlinkHref="#icon_sys_select"
            y="0" />
        </svg>
        选中设备
      </div>
      <div
        className="panel-body">
        <span
          title="">
          
        </span>
      </div>
    </div>
    <div
      className="panel panel-default panel-2">
      <div
        className="panel-heading">
        <svg>
          <use
            height="200"
            transform="scale(0.088,0.086)"
            viewBox="0 0 20 20"
            width="200"
            x="0"
            xlinkHref="#icon_device_operate"
            y="0" />
        </svg>
        设备操作
      </div>
      <div
        className="panel-body">
        <div>
          <span
            className="tit">
            设备开关：
          </span>
          <select
            className="form-control select "
            disabled={true}
            id="deviceSwitch"
            onChange={[Function]}
            value="">
            <option
              value="on">
              开启
            </option>
            <option
              value="off">
              关闭
            </option>
          </select>
          <button
            className="btn btn-primary"
            disabled={true}>
            应用
          </button>
        </div>
        <div>
          <span
            className="tit">
            调光：
          </span>
          <select
            className="form-control select "
            disabled={true}
            id="dimming"
            onChange={[Function]}
            value="" />
          <button
            className="btn btn-primary"
            disabled={true}>
            应用
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
